server/utils/hollaex-tools-lib/tools/user.js

```diff
@@ -4060,6 +4063,371 @@ const deleteAnnouncement = async (id) => {
+
+				basePermission = `/admin/${path.replace(/\./g, '/')}:${method}`;
+				permissionType = 'route';
+			}
+			else if (permission.startsWith('config:')) {
+				basePermission = permission.replace('config:', '');
+				permissionType = 'config';
+			}
+			else if (permission.startsWith('secret:')) {
+				basePermission = permission.replace('secret:', '');
+				permissionType = 'secret';
+			}
+			else {
+				throw new Error(`Invalid permission prefix: ${permission}. Must start with route:, config:, or secret:`);
+			}
+
+			validationGroups[permissionType].push(basePermission);
+			permissionsToStore.push(basePermission);
+		});
+
+		try {
+			if (validationGroups.route.length > 0) {
+				validateRoutePermissions(validationGroups.route);
+			}
+			if (validationGroups.config.length > 0) {
+				validateConfigPermissions(validationGroups.config);
+			}
+			if (validationGroups.secret.length > 0) {
+				validateSecretPermissions(validationGroups.secret);
+			}
+		} catch (error) {
+			throw new Error(`Permission validation failed: ${error.message}`);
+		}
+
+		const currentPermissions = role.permissions || [];
+		const permissionsChanged = (
+			currentPermissions.length !== permissionsToStore.length ||
+			!currentPermissions.every(p => permissionsToStore.includes(p))
+		);
+
+		if (permissionsChanged) {
+			updates.permissions = permissionsToStore;
+		}
+	}
+
+	if (Object.keys(updates).length > 0) {
+		const [affectedRows] = await Role.update(updates, {
+			where: { id: roleId }
+		});
+
+		if (affectedRows === 0) {
+			throw new Error('Update failed - no rows affected');
+		}
+	}
+
+	const updatedRole = await Role.findByPk(roleId);
+	return updatedRole;
+};
+
+const deleteExchangeUserRole = async (id) => {
+	const Role = getModel('role');
+	const role = await Role.findOne({
+		where: { id },
+	});
+
+	if (!role) {
+		throw new Error('Role not found');
+	}
+
+	if (role.users && role.users.length > 0) {
+		throw new Error('Cannot delete role with assigned users');
+	}
+
+	await role.destroy();
+	return { message: 'Role deleted successfully' };
+};
+
+const assignExchangeUserRole = async (user_id, role_id) => {
+	const Role = getModel('role');
+	const User = getModel('user');
+
+	const user = await User.findOne({ where: { id: user_id } });
+	if (!user) throw new Error('User not found');
+
+	const role = await Role.findOne({ where: { id: role_id } });
+	if (!role) throw new Error('Role not found');
+
+	user.role_id = role_id;
+	await user.save();
+
+	return user;
+};
+
+const fetchExchangeUserRole = async (user_id) => {
+	const User = getModel('user');
+	const user = await User.findOne({
+		where: { id: user_id },
+		include: [{
+			model: Role,
+			as: 'role',
+			attributes: ['id', 'name', 'permissions']
+		}],
+		attributes: ['id', 'email']
+	});
+
+	if (!user) throw new Error('User not found');
+
+	return {
+		user_id: user.id,
+		role: user.role
+	};
+};
+
+const hasPermission = async (user_id, permission) => {
```

Comment:
where is this function being used?

Reply:
yeah right. this was something i had initially, we don't need it anymore ill remove it